src/ast/scopes/Scope.ts

```diff
@@ -81,35 +81,37 @@ export default class Scope {
 		);
 	}
 
-	deshadow (names: Set<string>) {
+	deshadow (names: Set<string>, children = this.children) {
 		keys(this.variables).forEach(key => {
 			const declaration = this.variables[key];
 
 			// we can disregard exports.foo etc
 			if (declaration.exportName && declaration.isReassigned) return;
 
-			const name = declaration.getName(true);
-			let deshadowed = name;
+			let name = declaration.getName();
 
-			let i = 1;
+			if (!names.has(name)) {
+				return;
+			}
 
-			while (names.has(deshadowed)) {
+			name = declaration.name;
+			let deshadowed, i = 1;
+			do {
 				deshadowed = `${name}$$${i++}`;
```

Comment:
I think this could actually be changed to `${name}$${i++}` to avoid the extra "$". Might cause too much test noise for this PR, though, and might be better handled when/if we should decide to unify the deconflicting helpers. Just noticed.

Reply:
This is how it's always been it seems, so may result in large ecosystem churn to consider separately. The rule seems to work out as $$ for locals and $ for module specifiers with this method.

Reply:
(as in I agree!)